client/asset/xmr/process.go

```diff
@@ -0,0 +1,184 @@
+package xmr
+
+import (
+	"context"
+	"fmt"
+	"os/exec"
+	"path"
+	"runtime"
+	"strings"
+
+	"decred.org/dcrdex/dex"
+)
+
+const (
+	WalletServerRpcName = "monero-wallet-rpc"
+	WalletLogfileName   = WalletServerRpcName + ".log"
+	WalletLogLevel      = "2"
+	WalletFileName      = "dex"
+	WalletKeysFileName  = "dex.keys"
+)
+
+const (
+	CliName        = "monero-wallet-cli"
+	CliLogfileName = CliName + ".log"
+	CliLogLevel    = "3"
+)
+
+const (
+	HttpLocalhost                       = "http://127.0.0.1:"
+	Json2query                          = "/json_rpc"
+	MainnetWalletServerRpcPort          = "18083"
+	StagenetWalletServerRpcPort         = "38083"
+	DefaultRegtestWalletServerRpcPort   = "18083"
+	AlternateRegtestWalletServerRpcPort = "18087"
+)
+
+const (
+	DaemonAddressParam                = "--daemon-address="
+	RpcBindPortParam                  = "--rpc-bind-port="
+	StagenetParam                     = "--stagenet"
+	TrustedDaemonParam                = "--trusted-daemon"
+	WalletDirParam                    = "--wallet-dir="
+	WalletFileParam                   = "--wallet-file="
+	DisableRpcLoginParam              = "--disable-rpc-login"
+	AllowMismatchedDaemonVersionParam = "--allow-mismatched-daemon-version"
+	LogFileParam                      = "--log-file="
+	LogLevelParam                     = "--log-level="
+)
+
+const (
+	CliGenerateNewWalletParam       = "--generate-new-wallet"
+	CliMnemonicLanguageEnglishParam = "--mnemonic-language=English" // hard code
+	CliPasswordParam                = "--password="
+	CliPasswordFileParam            = "--password-file=" //TODO(xmr)
+	CliOfflineParam                 = "--offline"
+	CliCommandBalanceParam          = "--command=balance"
+	CliCommandAddressParam          = "--command=address"
+	CliCommandRefreshParam          = "--command=refresh"
+)
+
+func (r *xmrRpc) probeDaemon() error {
+	r.log.Trace("probeDaemon")
+	info, err := r.getInfo()
+	if err != nil {
+		return err
+	}
+	if info.Status != "OK" {
+		return fmt.Errorf("daemon bad status: %w - expected 'OK'", err)
+	}
+	r.daemonState.Lock()
+	r.daemonState.height = info.Height
+	r.daemonState.blockHash = info.TopBlockHash
+	r.daemonState.synchronized = info.Sychronized
+	r.daemonState.restricted = info.Restricted
+	r.daemonState.untrusted = info.Untrusted
+	r.log.Debugf("daemon %s -- height: %d, synchronized: %v, restricted: %v, untrusted: %v", r.daemonAddr,
+		r.daemonState.height, r.daemonState.synchronized, r.daemonState.restricted, r.daemonState.untrusted)
```

Comment:
Its probably better all the logs be sentences, or at least have some more info. This could start with "Probing daemon %s" so we know where it's being called.